File: parkkyungwon/codeforces/p2000/1366D Two Divisors.py
def get_spf(n: int):
    length = n + 1
    dp = bytearray(0 for _ in range(length))

    for i in range(2, int(length ** 0.5)):
        if dp[i]: continue

        for j in range(i * i, length, i):
            if dp[j]: continue
            
            dp[j] = i
    
    return dp


def main():
    input()
    arr = list(map(int, input().split()))
    answer1, answer2 = [], []
    spf = get_spf(10000000)

    for a in arr:
        factor = spf[a]
        divisor = a
        while not divisor % factor:
            divisor //= factor

        if divisor == 1:
            answer1.append(-1)
            answer2.append(-1)

        else:
            answer1.append(factor)
            answer2.append(divisor)
    
    print(*answer1)
    print(*answer2)


main()

File: parkkyungwon/codeforces/p2000/1366D Two Divisors.rs
use std::io::stdin;



fn get_sf() -> Vec<u32> {
    const LENGTH: usize = 10000001;
    let mut dp = vec![0u32; LENGTH];

    for i in 2..LENGTH {
        if dp[i] != 0 { continue }
        
        dp[i] = i as u32;

        for j in (i * i..LENGTH).step_by(i) {
            if dp[j] != 0 { continue }

            dp[j] = i as u32;
        }
    }

    dp
}


fn main() {
    let mut input = String::new();
    let _ = stdin().read_line(&mut input);
    input.clear();

    let _ = stdin().read_line(&mut input);
    let arr = input.split_whitespace()
    .map(|x| x.parse::<u32>().unwrap())
    .collect::<Vec<u32>>();

    let mut answer1: Vec<i32> = Vec::new();
    let mut answer2: Vec<i32> = Vec::new();

    let dp  = get_sf();

    for a in arr {
        let factor = dp[a as usize];
        let mut divisor = a;
        while divisor % factor == 0 { divisor /= factor }

        if divisor == 1 {
            answer1.push(-1);
            answer2.push(-1);
        } else {
            answer1.push(factor as i32);
            answer2.push(divisor as i32);
        }
    }

    for avec in [answer1, answer2] {
        println!("{} ", avec.iter( )
        .map(|x| x.to_string())
        .collect::<Vec<String>>()
        .join(" "));
    }
}
